### DISCUSSION TITLE
monterey + safari 16 + conditional ui

### DISCUSSION THREAD

**Discussion by siacomuzzi** [MasterKale/SimpleWebAuthn#242] [Troubleshooting]
Hi,

I just installed `Safari Technology Preview -  Release 150 (Safari 16.0, WebKit 17614.1.22.1.2)` in `macOS Monterey 12.4`.
I was trying to test the "Conditional UI" feature, but I just noted that `browserSupportsWebAuthnAutofill` helper method returns `false` because `window.PublicKeyCredential.isConditionalMediationAvailable()` is returning `false`.

Do you know if it is possible to enable this feature in Safari 16 for macOS Monterey?

Thank you

**Comment by siacomuzzi** [comment]
Same for `Chrome Canary 106.0.5226.0`, the `browserSupportsWebAuthnAutofill` helper method returns `false` because:

* `navigator.credentials.conditionalMediationSupported` is undefined, and
* `window.PublicKeyCredential.isConditionalMediationAvailable` is undefined

**Comment by MasterKale** [comment]
> Do you know if it is possible to enable this feature in Safari 16 for macOS Monterey?

On macOS conditional UI support is a combination of browser and OS unfortunately. I was miffed a bit when I realized I couldn't test it on Monterey even with access to Safari Tech Preview w/Safari 16 features, it _had_ to be on something running the Ventura beta.

> Same for `Chrome Canary 106.0.5226.0`, the `browserSupportsWebAuthnAutofill` helper method returns `false`

I left the following notes for myself over on example.simplewebauthn.dev and getting conditional UI working in Chrome Canary, maybe this'll help you:

```html
/**
 * Conditional UI test
 *
 * 1. Start Chrome Canary 105+ with the requisite Conditional UI flag:
 *
 * open -a /Applications/Google\ Chrome\ Canary.app --args --enable-features=WebAuthenticationConditionalUI
 *
 * 2. Create an entry in chrome://settings/passwords (temporary requirement) e.g.:
 *
 *   - Site: https://example.simplewebauthn.dev/
 *   - Username: user@example.simplewebauthn.dev
 *   - Password: whatever
 *
 * 3. Register a credential
 *
 * 4. Reload the page
 *
 * 5. Interact with the username field above the Authenticate button
 *
 * Notes:
 *
 * I'm currently trying to get to calling WebAuthn as fast as I can here, there's a
 * Chrome race condition with autofill that sometimes prevents a credential from appearing.
 *
 * See: https://bugs.chromium.org/p/chromium/issues/detail?id=1322967&q=component%3ABlink%3EWebAuthentication&can=2
 *
 * I've been assured this race condition is temporary, at which point we'll probably be able
 * to include this just before </body> as we'd typically do. And at that point we can
 * probably use async/await as well for more sane-looking code.
 */
```

**Comment by siacomuzzi** [comment]
Great, I'll use this workaround in Chrome Canary until I've installed Ventura.

<img width="446" alt="image" src="https://user-images.githubusercontent.com/178506/183471014-a6332b66-c420-47a9-83ea-4c496005c195.png">

Thank you Matthew!
